### ISSUE TITLE
Incorrect isgraphical

### ISSUE THREAD

**Issue by abhinavmehndiratta** [JuliaGraphs/LightGraphs.jl#1153]
```julia
julia> a = [1,3,3,3]
4-element Array{Int64,1}:
 1
 3
 3
 3

julia> isgraphical(a)
true
```
This should return false.
Erdos-Gallai theorem requires the degrees are in descending order.The function does not check that.
Example - 
```julia
julia> a = [3,3,3,1]
4-element Array{Int64,1}:
 3
 3
 3
 1

julia> isgraphical(a)
false
```

**Comment by adamglos92** [comment]
Another thing is that the algorithm can be improved in complexity
```
function isgraphical_1(degs::Vector{Int})
    iseven(sum(degs)) || return false
    n = length(degs)
    cur_sum = 0
    mindeg = (i->min(i, degs[i])).(degs)
    cum_min = sum(mindeg)
    for r = 1:(n - 1)
        cur_sum += degs[r]
        cum_min -= mindeg[r]
        cond = cur_sum <= r * (r - 1) + cum_min
        cond || return false
    end
    return true
end

v = fill(99999, 100000)
isgraphical_1(v)
isgraphical(v)
@time isgraphical(v)
@time isgraphical_1(v)
```

Output:
```
  6.676736 seconds (200.00 k allocations: 4.578 MiB)
  0.000446 seconds (6 allocations: 781.484 KiB)
```
I haven't included sorting in order to compare it with the original definition.

**Comment by adamglos92** [comment]
Please check the correctness of my code :)

**Comment by abhinavmehndiratta** [comment]
This seems correct, I was about to suggest the idea of storing prefix sum array initially instead of calculating each time.

**Comment by adamglos92** [comment]
If I understand You correctly, you proposed to precompute the prefix sums with function similar to `cumsum`. Precomputing with `cumsum` requires more memory and is inefficient if the output is false, especially for small `r`.

**Comment by abhinavmehndiratta** [comment]
> If I understand You correctly, you proposed to precompute the prefix sums with function similar to `cumsum`. Precomputing with `cumsum` requires more memory and is inefficient if the output is false, especially for small `r`.

Yes,I realized your way of doing it is better :sweat_smile: .

**Comment by abhinavmehndiratta** [comment]
Closing this as fixed by #1155
